cloudinit/config/cloud-init-schema.json

```diff
@@ -543,6 +543,244 @@
           }
         }
       }
+    },
+    "cc_package_update_upgrade_install": {
+        "type": "object",
+        "properties": {
+          "packages": {
+            "type": "array",
+            "description": "A list of packages to install. Each entry in the list can be either a package name or a list with two entries, the first being the package name and the second being the specific package version to install.",
+              "items": {
+                "oneOf": [
+                  {"type": "array", "items": {"type": "string"}, "minItems": 2, "maxItems": 2},
+                  {"type": "string"}
+                ]
+              },
+              "minItems": 1
+            },
+            "package_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to update packages. Happens before upgrade or install. Default: ``false``"
+            },
+            "package_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to upgrade packages. Happens before install. Default: ``false``"
+            },
+            "package_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "Set ``true`` to reboot the system if required. Default: ``false``"
+            },
+            "apt_update": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_update``. Default: ``false``"
+            },
+            "apt_upgrade": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_upgrade``. Default: ``false``"
+            },
+            "apt_reboot_if_required": {
+                "type": "boolean",
+                "default": false,
+                "description": "DEPRECATED. Alias for ``package_reboot_if_required``. Default: ``false``"
+            }
```

Comment:
Curious if we should just remove these from the schema altogether as a means of deprecation. Since we don't strictly enforce schemas when cloud-init runs, they'll still work but we'd see a schema validation error in the logs.

Reply:
It might be more user friendly to have a separate message for users when a config is invalid vs when they are using a deprecated key. I'd lean towards leaving it how you have it for now.

Related: a quick test with your code shows that removing these deprecated keys from the jsonschema doesn't cause warnings with `cloud-init devel schema --config-file file.yml`. I think we want errors with the cli command as well as with the cloud-init runtime validation warning, so I assume that will take some more work.

My test:
```
root@my-host:~# cloud-init devel schema --config-file tmp.yml 
Valid cloud-config: tmp.yml
root@my-host:~# cat tmp.yml 
#cloud-config
packages:
  - pwgen
  - pastebinit
  - [libpython3.8, 3.8.10-0ubuntu1~20.04.2]
apt_update: true
apt_upgrade: true
apt_reboot_if_required: true

# demonstrating that the keys do not exist in jsonschema
root@my-host:~# grep -e apt_update -e apt_upgrade -e apt_reboot_if_required /usr/lib/python3/dist-packages/cloudinit/config/cloud-init-schema.json

# demonstrating that this instance is using the new schema definition
root@my-host:~# grep -e apt_update -e apt_upgrade -e apt_reboot_if_required -e cc_package_update_upgrade_install /usr/lib/python3/dist-packages/cloudinit/config/cloud-init-schema.json
    "cc_package_update_upgrade_install": {
    { "$ref": "#/$defs/cc_package_update_upgrade_install" },

```

Reply:
> Related: a quick test with your code shows that removing these deprecated keys from the jsonschema doesn't cause warnings with cloud-init devel schema --config-file file.yml. I think we want errors with the cli command as well as with the cloud-init runtime validation warning, so I assume that will take some more work.

This is a more general comment that would apply to all schemas and not something to fix here, correct? Just making sure I understand correctly.

Reply:
> > Related: a quick test with your code shows that removing these deprecated keys from the jsonschema doesn't cause warnings with cloud-init devel schema --config-file file.yml. I think we want errors with the cli command as well as with the cloud-init runtime validation warning, so I assume that will take some more work.
> 
> This is a more general comment that would apply to all schemas and not something to fix here, correct? Just making sure I understand correctly.

Correct, I don't think this is a local problem. I just happened to be using your PR to see what kind of error messages I would get and was surprised I didn't see any. We can do that in a separate PR.

Reply:
I agree that feature for rendering deprecation warnings is desirable for this current epic of work, but not for this specific PR.  I added this [work item](https://warthogs.atlassian.net/browse/SC-833)